metricbeat/helper/prometheus/prometheus.go

```diff
@@ -161,6 +174,84 @@ func (p *prometheus) GetProcessedMetrics(mapping *MetricsMapping) ([]common.MapS
 			}
 
 			if field != "" {
+				// TODO only if expanding values
+				if v, ok := value.(common.MapStr); ok && p.EventLayout == ExpandedBucketsLayout {
+					var expanded common.MapStr
+					// only checking "sum", not "count", since they are populated
+					// checking "sum" existence only at commonMetric.GetValue function
+					if _, ok := v["sum"]; ok {
+						expanded = common.MapStr{}
+						expanded["sum"] = v["sum"]
+						expanded["count"] = v["count"]
+
+						// create unique selector for the event
```

Comment:
This looks like 3 times almost the indentical code? Could it be put into a method?